### PR TITLE
Add support for Sophos Home and Sophos Hitman

### PR DESCRIPTION
**Pull request by cecinestpasunepipe**
(DIS-1525)

### CODE REVIEW

File: dissect/target/plugins/apps/av/sophos.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            details (string): Detailed description of the alert.
+
+        Note that because Hitman also catches suspicious behaviour of
+        systems, the details field might contain a lot of text, it might
+        contain stracktraces etc.
+        """
+        try:
+            fh = self.target.fs.path(self.LOG_SOPHOS_HITMAN).open("rb")
+            db = sqlite3.SQLite3(fh)
+            alerts = list(filter(lambda t: t.name == "Alerts", db.tables()))[0]
```
Comment (pyrco) [inline dissect/target/plugins/apps/av/sophos.py:70]: The `Alerts` table is always present even if there have been no alerts?

File: dissect/target/plugins/apps/av/sophos.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            self.target.log.error(f"Error occurred during reading alerts: {error}.")
+
+    @export(record=SophosLogRecord)
+    def sophoshomelogs(self) -> Iterator[SophosLogRecord]:
+        """Return log history records from Sophos Home.
+
+        Yields SophosLogRecord with the following fields:
+            ts (datetime): Timestamp.
+            description (string): Short description of the alert.
+            path (string): Path to the infected file (if available).
```
Comment (pyrco) [inline dissect/target/plugins/apps/av/sophos.py:89]:
```suggestion
            path (path): Path to the infected file (if available).
```

At least according to the record definition ;-)

File: dissect/target/plugins/apps/av/sophos.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        log = self.target.fs.path(self.LOG_SOPHOS_HOME).open("rt", 0, "utf-16le")
+        while line := log.readline():
+            if line.find(self.MARKER_INFECTION) > -1:
+                try:
+                    ts, json_data = line.split(" ", maxsplit=2)
+                    details = json.loads(json_data)
+                    yield SophosLogRecord(
+                        ts=ts,
+                        description=details.get("threat_name", details),
+                        path=details.get("targets", [{}])[0].get("file_path", None),
```
Comment (pyrco) [inline dissect/target/plugins/apps/av/sophos.py:101]:
```suggestion
                        path=details.get("targets", [{}])[0].get("file_path", None),
```
I'd rather have this split out and in multiple lines for readability.
Also it should be cast to a flow.record path type.

**Review by pyrco — CHANGES_REQUESTED** [review]
Maybe a slightly different commit message:

> Add plugins for Sophos Home and Sophos Hitman

File: dissect/target/plugins/apps/av/sophos.py (hunk 1 above)
Comment (cecinestpasunepipe) [reply dissect/target/plugins/apps/av/sophos.py:72]: Yes it's a fixed database schema.

File: dissect/target/plugins/apps/av/sophos.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        log = self.target.fs.path(self.LOG_SOPHOS_HOME).open("rt", 0, "utf-16le")
+        while line := log.readline():
+            if line.find(self.MARKER_INFECTION) > -1:
+                try:
+                    ts, json_data = line.split(" ", maxsplit=2)
+                    details = json.loads(json_data)
+
+                    path_to_infected_file = None
+                    if targets := details.get("targets", None):
+                        if len(targets) > 0:
```
Comment (pyrco) [inline dissect/target/plugins/apps/av/sophos.py:103]: Is this check needed? An empty list will already resolve to `False` in the line above.

**Review by pyrco — APPROVED** [review]

**Pull request merged**